### PR TITLE
[S4] [User Story] Set up Github Actions for CI

### PR DESCRIPTION
**Pull request by emwangs**
Sets up linting checks for CPP

### CODE REVIEW

File: ci/clang-format-checker.sh
Comment (evanmwilliams) [inline ci/clang-format-checker.sh:1]: I think you can rename this to `clang-format-checker.py`

**Review by evanmwilliams — COMMENTED** [review]
Made one comment -lgtm!

File: ci/clang-format-checker.sh
Comment (emwangs) [reply ci/clang-format-checker.sh:1]: i don't want to

**Review by evanmwilliams — APPROVED** [review]
lgtm i guess...

**Pull request merged**